packages/backend-deployer/src/cdk_error_mapper.ts

```diff
@@ -229,6 +229,14 @@ export class CdkErrorMapper {
       errorName: 'LambdaMaxSizeExceededError',
       classification: 'ERROR',
     },
+    {
+      errorRegex:
+        /InvalidParameterValueException: Uploaded file must be a non-empty zip/,
+      humanReadableErrorMessage: 'Lambda bundled into an empty zip',
+      resolutionMessage: `Try removing '.amplify/artifacts' then running the command again. If it still doesn't work, see https://github.com/aws/aws-cdk/issues/18459 for more methods.`,
+      errorName: 'LambdaEmptyZipError',
+      classification: 'ERROR',
```

Comment:
From what I have seen in the related issue, it's not a customer issue but the library one (either us or CDK). It's ok to parse and provide resolution steps here, but we should mark this as a `FAULT`

Reply:
Got it, I saw some root causes are user error (gitignoring function code, pointing to different file/directory which is empty, etc) but yeah it makes sense we should mark this as a `FAULT`.